### DISCUSSION TITLE
Upgrade from 0.9.4 SyngAgent method

### DISCUSSION THREAD

**Discussion by carlosxjose** [Mimetis/Dotmim.Sync#821] [Q&A]
After 0.9.4 Release the **SyncAgent** method only allow 3 parameters and the Agent method doesn't has the **Parameters** for filtering anymore...

```
// Creating an agent that will handle all the process
var agent = new SyncAgent(clientProvider, serverProvider, syncOptions, syncSetup);

if (!agent.Parameters.Contains("inv_master_sts_nulo"))
  agent.Parameters.Add("inv_master_sts_nulo", 0);
```

I am still on testing mode but I always download the latest releases to test. I know you are merging all the commits and fixes. I am planning to migrate my solution from SQLServer 2012 Replication with SQL CE to Dotmim SYNC.... I've been testing a lot of different librarys and frameworks and the only with 1000% owsome performance and similar is this.

Thanks for all your work.

**Comment by Mimetis** [comment]
You Should go for latest version 0.9.7
Lot of changes from 0.9.4
Try it on New databases
Read the doc. Parameters Should now be passed as argument to SynchronizeAsync method.

**Comment by carlosxjose** [comment]
Great!!! That was what I was wondering.... You just refactored to setup the parameters once.

I've been testing this week the 0.9.7.

**Reply by Mimetis** [reply]
And you can use différent parameters for the same client Database

**Comment by sylverbee** [comment]
I have same pb : migration from 0.9.4 to 0.9.7 raises a lot of constrains and rework on code.
I miss documentation to help on doing so

**Reply by Mimetis** [reply]
did you checked https://github.com/Mimetis/Dotmim.Sync/discussions/802 ?
